wandb/deploys/launch-agent.yaml

```diff
@@ -0,0 +1,86 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: wandb-launch-agent
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: launch-agent-demo
+  namespace: wandb-launch-agent
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: launch-agent-demo
+  template:
+    metadata:
+      labels:
+        app: launch-agent-demo
+    spec:
```

Comment:
We should attach a service account to this deployment and ensure it has the necessary permissions to do what we need to do.  That would mean adding `serviceAccountName: wandb-launch-agent` here in the spec, and creating the following resources:

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: wandb-launch-agent
---
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: wandb-launch-agent
rules:
- apiGroups: [""]
  resources: ["pods", "pods/log"]
  verbs: ["get", "list", "watch"]
- apiGroups: ["batch", "extensions"]
  resources: ["jobs"]
  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: wandb-launch-agent
subjects:
- kind: ServiceAccount
  name: wandb-launch-agent
  namespace: wandb-launch-agent
roleRef:
  kind: ClusterRole
  name: wandb-launch-agent
  apiGroup: rbac.authorization.k8s.io
```

We may need more permissions than the one's I've specified here.  I would also imagine users would want to customize what namespaces our agent could launch stuff in in which case we would need a `Role` instead of a `ClusterRole` in the appropriate namespace.  I suppose a more sane default would be to let the user put our launch agent into an existing namespace they have and assume we only want permissions to create / read the various resources in that namespace...

Reply:
Sent you an updated yaml with the service account that will be merged in briefly, for now I've pulled the yaml from this PR